### PR TITLE
UCS/TEST: Add arbiter data structure.

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com:8888/jenkins/job/gh-ucx-pr/869/ for details.

**Comment by shamisp** [comment]
autobot build

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com:8888/jenkins/job/gh-ucx-pr/871/ for details.

File: src/ucs/datastruct/list.h [hunk 1]
```diff
@@ -36,6 +36,22 @@ static inline void ucs_list_head_init(ucs_list_link_t *head)
     head->prev = head->next = head;
 }
 
+/* Insert an element in-between to list elements. Any elements which were in this
```
Comment (shamisp) [inline src/ucs/datastruct/list.h]: /** - doxygen format

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/294/ for details. (Mellanox internal link)

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/296/ for details. (Mellanox internal link)

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/297/ for details. (Mellanox internal link)

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com:8888/jenkins/job/gh-ucx-pr/872/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com:8888/jenkins/job/gh-ucx-pr/873/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com:8888/jenkins/job/gh-ucx-pr/874/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com:8888/jenkins/job/gh-ucx-pr/875/ for details.

**Comment by mellanox-github** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com:8888/jenkins/job/gh-ucx-pr/880/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/298/ for details. (Mellanox internal link)

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/300/ for details. (Mellanox internal link)

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/302/ for details. (Mellanox internal link)

**Comment by shamisp** [comment]
:+1

**Comment by yosefe** [comment]
:+1:

**Pull request merged**